controller/api/destination/endpoint_translator.go

```diff
@@ -257,10 +257,7 @@ func (et *endpointTranslator) sendClientAdd(set watcher.AddressSet) {
 			err         error
 		)
 		if address.Pod != nil {
-			opaquePorts, err = getAnnotatedOpaquePorts(address.Pod, et.defaultOpaquePorts)
-			if err != nil {
-				et.log.Errorf("failed to get opaque ports for pod %s/%s: %s", address.Pod.Namespace, address.Pod.Name, err)
-			}
+			opaquePorts = getAnnotatedOpaquePorts(address.Pod, et.defaultOpaquePorts)
 			wa, err = createWeightedAddr(address, opaquePorts, et.enableH2Upgrade, et.identityTrustDomain, et.controllerNS, et.log)
```

Comment:
Maybe it's outside the scope of this change, but we could also change `createWeightedAddr` to accept a slice of PortRanges instead of a set of ports.  It needs to be able to determine if a given port is opaque and it could do this by checking to see if the port is contained in any of the opaque port ranges.  On the other hand, it works fine today with an expanded set of ports so maybe there's not a good reason to change it. 🤷‍♀️

Reply:
I might address this in a follow-up, if that's okay?